store/bolt/user_store.go

```diff
@@ -0,0 +1,124 @@
+package bolt
+
+import (
+	"errors"
+
+	"github.com/boltdb/bolt"
+	"github.com/gopheracademy/congo"
+	"github.com/gopheracademy/congo/store/bolt/internal"
+)
+
+var (
+	// ErrUserNotFound is returned when mutating a user that doesn't exist.
+	ErrUserNotFound = errors.New("user not found")
+
+	// ErrEmailRequired is returned when creating a user without an email.
+	ErrEmailRequired = errors.New("email address required")
+)
+
+// UserStore represents the data store for user data.
+type UserStore struct {
+	db *bolt.DB
+}
+
+// initialize creates the top-level buckets for the store.
+func (s *UserStore) init() error {
+	return s.db.Update(func(tx *bolt.Tx) error {
+		tx.CreateBucketIfNotExists([]byte("users"))
+		tx.CreateBucketIfNotExists([]byte("users.email"))
+		return nil
+	})
+}
```

Comment:
This creates our buckets on startup so we don't have to check for them later. They're guaranteed to be there after this.